beamer/agent/middleware.py

```diff
@@ -27,9 +53,12 @@ def _result_ok(response: RPCResponse) -> bool:
 # further requests specifying a concrete block number can be satisfied from
 # the cache.
 def cache_get_block_by_number(
-    make_request: Callable[[RPCEndpoint, Any], RPCResponse], _w3: Web3
+    make_request: Callable[[RPCEndpoint, Any], RPCResponse], _w3: Web3, chain_id: ChainId
```

Comment:
nitpick: 

with 

```py
def partial_cache(middleware, chain_id: ChainId) -> Callable[[RPCEndpoint, Any], RPCResponse]:
    if chain_id not in _BLOCK_STORAGE:
        _BLOCK_STORAGE[chain_id] = _BlockCache()

    cache = _BLOCK_STORAGE[chain_id]

    return partial(middleware, cache=cache)
```

we could convert the chain id agrument into the cache object